python/ray/data/impl/output_buffer.py

```diff
@@ -0,0 +1,71 @@
+from typing import Callable, Any, Optional
+
+from ray.data.block import Block
+from ray.data.impl.arrow_block import DelegatingArrowBlockBuilder
+
+
+class BlockOutputBuffer(object):
+    """Generates output blocks of a given size given a stream of inputs.
+
+    This class is used to turn a stream of items / blocks of arbitrary size
+    into a stream of blocks of ``target_max_block_size``. The caller should
+    check ``has_next()`` after each add call, and call ``next()`` to get the
```

Comment:
Nit: Will probably need to fix the 80 character width overflow here.
```suggestion
    check ``has_next()`` after each ``add()`` call, and call ``next()`` to get the
```